robot_smach_states/src/robot_smach_states/human_interaction/give_directions.py

```diff
@@ -10,40 +13,46 @@
 
 # Robot Smach States
 from robot_smach_states.navigation.navigate_to_symbolic import NavigateToSymbolic
+from robot_smach_states.util.designators.ed_designators import EntityByIdDesignator
 
 
 class GiveDirections(smach.State):
     """
     Robot tells the operator how to get to a certain entity
     """
-    def __init__(self, robot, entity_designator, radius=1.5):
-        # type: (Robot, any) -> any
+    def __init__(self, robot, entity_designator, x_threshold=0.75, y_threshold=1.5):
+        # type: (Robot, EntityByIdDesignator, float, float) -> None
         """
         Init
-
-        :param robot: (Robot) API object
-        :param entity_designator: (EdEntityDesignator) resolving to the entity the operator wants to go to
-        :param radius: passing closer to an entity than this radius, the entity will be mentioned
+        :param robot: API object
+        :type robot: Robot
+        :param entity_designator: resolving to the entity the operator wants to go to
+        :type entity_designator: EdEntityDesignator
+        :param x_threshold: if the entity is closer than this distance in x-direction w.r.t. the path frame
+        it is considered 'passed'
+        :type x_threshold: float
+        :param y_threshold: if the entity is closer than this distance in y-direction w.r.t. the path frame
+        it is considered 'passed'
+        :type y_threshold: float
         """
 
         super(GiveDirections, self).__init__(outcomes=["succeeded", "failed"])
 
         self._robot = robot
         self._entity_designator = entity_designator
-        self._radius = radius
-
-    def execute(self, ud):
+        self._x_threshold = x_threshold
+        self._y_threshold = y_threshold
 
+    def execute(self, userdata=None):
         # Get the constraints for the global planner
+        nav_constraints = dict()
         for area in ["in_front_of", "near"]:
-            nav_contraints = NavigateToSymbolic.generate_constraint(
+            nav_constraints[area] = NavigateToSymbolic.generate_constraint(
```

Comment:
@MatthijsBurgh jouw wijziging maakt dit idd iets robuuster (al ga ik ervan uit dat je typisch wel een pad vindt). Ik zou er wel een Ordereddict van maken zodat je voorkeur geeft aan in front of. Anders kun je met near verkeerd uitkomen

Reply:
Done.
(Ik vond geen path bij de workbench. De in_front_of, zit daar onder een van de operator tables)